lib/statsd/instrument/environment.rb

```diff
@@ -39,26 +68,65 @@ def datagram_builder_class
   #
   # @return [String] The detected environment.
   def environment
-    if defined?(Rails) && Rails.respond_to?(:env)
+    if env['STATSD_ENV']
+      env['STATSD_ENV']
+    elsif defined?(Rails) && Rails.respond_to?(:env)
       Rails.env.to_s
     else
-      ENV['RAILS_ENV'] || ENV['RACK_ENV'] || ENV['ENV'] || 'development'
+      env['RAILS_ENV'] || env['RACK_ENV'] || env['ENV'] || 'development'
     end
   end
 
-  # Sets default values for sample rate and logger.
-  #
-  # - Default sample rate is set to the value in the STATSD_SAMPLE_RATE environment variable,
-  #   or 1.0 otherwise. See {StatsD#default_sample_rate}
-  # - {StatsD#logger} is set to a logger that send output to stderr.
-  #
-  # If you are including this library inside a Rails environment, additional initialization will
-  # be done as part of the {StatsD::Instrument::Railtie}.
-  #
-  # @return [void]
-  def setup
-    StatsD.default_sample_rate = ENV.fetch('STATSD_SAMPLE_RATE', 1.0).to_f
-    StatsD.logger = Logger.new($stderr)
+  def statsd_implementation
+    env.fetch('STATSD_IMPLEMENTATION', 'statsd')
+  end
+
+  def statsd_sample_rate
+    env.fetch('STATSD_SAMPLE_RATE', 1.0).to_f
+  end
+
+  def statsd_prefix
+    env.fetch('STATSD_PREFIX', nil)
```

Comment:
I thought you were planning on getting rid of the prefix?

Reply:
I don't think we can, because it is used all over the place. Most applications set a prefix to not interfere with each other.